stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/view/MessageListView.kt

```diff
@@ -589,20 +596,31 @@ public class MessageListView : ConstraintLayout {
         buffer.enqueueData(listItem)
     }
 
-    private fun handleNewWrapper(listItem: MessageListItemWrapper) {
-        buffer.hold()
-
-        val isThreadStart = !adapter.isThread && listItem.isThread
-        val isOldListEmpty = adapter.currentList.isEmpty()
+    public fun setMessageListItemFilter(messageListItemFilter: MessageListItemFilter) {
+        check(::adapter.isInitialized.not()) { "Adapter was already initialized, please set MessageListItemFilter first" }
+        this.messageListItemFilter = messageListItemFilter
+    }
 
-        adapter.isThread = listItem.isThread
-        adapter.submitList(listItem.items) {
-            scrollHelper.onMessageListChanged(
-                isThreadStart = isThreadStart,
-                hasNewMessages = listItem.hasNewMessages,
-                isInitialList = isOldListEmpty && listItem.items.isNotEmpty()
-            )
-            buffer.active()
+    private fun handleNewWrapper(listItem: MessageListItemWrapper) {
+        CoroutineScope(DispatcherProvider.IO).launch {
+            messageListItemFilter.filterMessageListItem(listItem.items).let { filteredList ->
```

Comment:
`let` is fancy, but is there any advantage of using it here instead of just `val filteredList = ...`? It seems to increase the nesting and complexity of the code for no gain.

Reply:
Good point